web3/providers/ipc.py

```diff
@@ -198,7 +199,8 @@ def __init__(self, ipc_path=None, testnet=False, timeout=10, *args, **kwargs):
         self._socket = PersistantSocket(self.ipc_path)
         super().__init__(*args, **kwargs)
 
-    def make_request(self, method, params):
+    async def make_request(self, method, params):
+        await asyncio.sleep(0)
```

Comment:
Perhaps a comment explaining why this sleep is needed. 
Wondering if this code is blocking? Does IPCProvider return control to mainloop when it is busy on a IO intensive task? Asking because you got by with just adding a sleep.

Reply:
This is a placeholder.  I will be working on making the ipc socket IO non-blocking in this PR next.